text/0000-fabric-private-chaincode-1.0.md

```diff
@@ -0,0 +1,720 @@
+It is implemented as a chaincode which maintains a list of all Chaincode Enclaves deployed on the peers in a channel.
+The registry associates with each enclave their identity, associated public keys and an attestation linking them.
+In detail, after an FPC Chaincode definition is committed on the channel, the chaincode's hosting enclave must be registered by invoking the Enclave Registry, in order to become operational.
+When registering a new enclave, the enclave registry verifies the provided enclave attestation (signed by the trusted hardware manufacturer) and checks it against the chaincode definition. In particular, the enclave attestation contains the code identity (`MRENCLAVE`) that must match the `version` field of the chaincode definition (recall deployment section earlier).
+If the verification succeeds, the enclave registry stores the attested data (including the public keys) on the ledger and thereby make it available for members of the channel.
+<!--  -->
+This enables any channel member to inspect an enclave attestation to verify the genuinity of the enclave in order to establish trust in its public keys.
+
+Additionally, the registry manages chaincode specific keys, including a chaincode public encryption key, and facilitates corresponding key-management among authorized chaincode enclaves.
+The registry is particularly relevant for clients, for retrieving the encryption key to protect the transaction arguments and set up a direct secure channel.
+Note that clients can access the enclave registry using "strong queries" where multiple peers are queried in order to get a correct response and not relying on a single peer.
+This prevents a malicious peer from responding with a wrong transaction encryption key.
+
+
+
+### Enclave Endorsement Validation
+The Enclave Endorsement Validation component verifies the correctness of a result of an FPC Chaincode execution and persists state updates to the ledger.
+In particular, the Validation Logic receives the output of a FPC Chaincode invocation, which is encrypted and signed by the enclave.
+The validation logic verifies the signature over the enclave execution response and that the response was produces by an enclave registered at the  Enclave Registry.
+Once the verification succeeds, the Enclave Endorsement Validation component applies any state updates issued by the FPC Chaincode.
+As the Validation logic is bundled together with the FPC Chaincode in a single Fabric chaincode package,
+these updates are eventually committed within the same namespace.
+Hence, they will be visible to the FPC Chaincode in subsequent invocations.
+
+
+
+## Deployment Process
+
+This section details the turn-up process for a FPC Chaincode.
+The normal Fabric chaincode deployment is extended with the creation and registration of a chaincode enclave as illustrated in the figure below.
+
+![Deployment](../images/fpc/high-level/Slide3.png)
```

Comment:
In diagram:
Typo in Step 1 "aggres".
Two Step 12s.

Reply:
Fixed